pkg/common/runner/pod.go

```diff
@@ -152,15 +156,40 @@ func (r *Runner) createPod(ctx context.Context) (pod *kubev1.Pod, err error) {
 	// setup git repos to be cloned in init containers
 	r.Repos.ConfigurePod(&pod.Spec)
 
-	// retry until Pod can be created or timeout occurs
-	var createdPod *kubev1.Pod
-	err = wait.PollUntilContextTimeout(ctx, fastPoll, podCreateTimeout, false, func(ctx context.Context) (done bool, err error) {
-		if createdPod, err = r.Kube.CoreV1().Pods(r.Namespace).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
-			r.Error(err, fmt.Sprintf("error creating %s/%s runner Pod", r.Namespace, r.Name))
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.Name,
+		},
+		Spec: batchv1.JobSpec{
+			Template: kubev1.PodTemplateSpec{
+				Spec: pod.Spec,
+			},
+		},
+	}
+	// retry until Job can be created or timeout occurs
+	if err = wait.PollUntilContextTimeout(ctx, fastPoll, podCreateTimeout, false, func(ctx context.Context) (done bool, err error) {
+		if job, err = r.Kube.BatchV1().Jobs(r.Namespace).Create(ctx, job, metav1.CreateOptions{}); err != nil {
+			r.Error(err, fmt.Sprintf("error creating %s/%s runner Job", r.Namespace, r.Name))
 		}
 		return err == nil, nil
-	})
-	return createdPod, err
+	}); err != nil {
+		return nil, err
+	}
+	r.UID = job.UID
```

Comment:
Does the job have a UID after creation or do we need to create a new UUID for it? https://pkg.go.dev/k8s.io/apimachinery/pkg/util/uuid#NewUUID

I'd also be surprised if we still need the ownerreference stuff since we are creating a job now. Were you able to test this out locally?